packages/dapp/src/components/Controlbar/Controlbar.tsx

```diff
@@ -76,6 +76,14 @@ const Controlbar = ({ vcs, isConnected }: ControlbarProps) => {
         newVcs.push(finalVC);
       });
       changeVcs([...vcs, ...newVcs]);
+
+      const queryResult = await api.queryVCs();
+      if (isError(queryResult)) {
+        return false;
+      }
+      if (queryResult.data) {
+        useSnapStore.getState().changeVcs(queryResult.data);
```

Comment:
Can you check and refactor to use the store functions more consistently. 
Like this: [link](https://github.com/pmndrs/zustand#selecting-multiple-state-slices)
And only use this: [link](https://github.com/pmndrs/zustand#readingwriting-state-and-reacting-to-changes-outside-of-components) where needed ?